### ISSUE TITLE
Syncthing ignores x-forwarded-proto causing redirect loop when reverse proxied

### ISSUE THREAD

**Issue by optix2000** [syncthing/syncthing#4860]
The [docs](https://docs.syncthing.net/users/reverseproxy.html) lay out some steps for setting up a reverse proxy in front of Syncthing. However, when I try to set it up, I get an infinite loop of https upgrade redirects when Syncthing is set to use HTTPS.

Using the following config for nginx:
```
location ^~ /syncthing/ {

  proxy_set_header Host $host;
  proxy_set_header X-Real-IP $remote_addr;
  proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
  proxy_set_header X-Forwarded-Proto $scheme;
  proxy_read_timeout 600s;
  proxy_send_timeout 600s;
  proxy_pass http://127.0.0.1:8384/;
}

```
And default configs for Syncthing, except for `Use HTTPS for GUI` checked, and `Insecure Skip Hostcheck` checked, to allow for non localhost domains while bound to localhost.

I see that nginx is sending all the headers properly, however Syncthing will ignore `X-Forwarded-Proto`, and still attempt the https redirect.
```
23:52:48.548830 IP localhost.60158 > localhost.8384: Flags [P.], seq 1:799, ack 1, win 11, options [nop,nop,TS val 690779790 ecr 690779790], length 798
E..R{.@.@............. ....<_........G.....
),v.),v.GET / HTTP/1.0
Host: example.com
X-Real-IP: 73.
X-Forwarded-For: 73.
X-Forwarded-Proto: https
Connection: close
pragma: no-cache
cache-control: no-cache
authorization: Basic 
upgrade-insecure-requests: 1
accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8
accept-encoding: gzip, deflate, br
accept-language: en-US,en;q=0.9

23:52:48.549110 IP localhost.8384 > localhost.60158: Flags [P.], seq 1:330, ack 799, win 12, options [nop,nop,TS val 690779790 ecr 690779790], length 329
E..}.p@.@........... ..._......Z.....q.....
),v.),v.HTTP/1.0 307 Temporary Redirect
Content-Type: text/html; charset=utf-8
Location: https://example.com/
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
X-Xss-Protection: 1; mode=block
Date: Wed, 04 Apr 2018 06:52:48 GMT
Content-Length: 63

<a href="https://example.com/">Temporary Redirect</a>.
```

### Version Information

Syncthing Version: v0.14.45
OS Version: Ubuntu 16.04

If there's no intention to support `x-forwarded-proto` in the near future, there should probably be some notes added to the docs signifying the https redirect needs to be done on the reverse proxy if one is being used.

**Comment by AudriusButkevicius** [comment]
As the issue template explains, please use the [forum](https://forum.syncthing.net/) for support.
Please use the search on the forum and the closed issue tracker for simillar issues before starting a new thread. 
If this issue is later proven to be a bug, we will reopen this issue or open a new one.

**Comment by optix2000** [comment]
@AudriusButkevicius I did read the issue template, and I believe this falls under a bug report. If reverse proxying _is_ currently supported, as`X-Forward-Proto` appears to not work (as outlined in the issue). If reverse proxying _is not_ currently supported, the documentation should be changed to reflect that.

Yes, arguably this could be a forum discussion topic first, but the template explicitly says "Please do post actual bug reports and feature requests.", and I believe in both cases this would be considered a bug, as outlined above.

> Please use the search on the forum and the closed issue tracker for simillar issues before starting a new thread.
I did both, and I could not find any issues or forum topics that were related to this exact issue (as it's also rather broad). Is there an issue/forum topic that's a duplicate of this?

**Comment by calmh** [comment]
As you are sending HTTP requests and not HTTPS, I check the checkbox to enforce HTTPS.

**Comment by AudriusButkevicius** [comment]
Or force your nginx to forward to https, as you force https in syncthing, and forward to http, which makes perfect sense to end up a redirect loop.

**Comment by optix2000** [comment]
I think there's some misunderstandings here (yes the issue title could have been better).

As I mentioned in the issue, [the docs](https://docs.syncthing.net/users/reverseproxy.html) give examples of how to set up a reverse proxy in front of Syncthing.
This implies that Syncthing supports having reverse proxies set up in front of it.

The issue here is that there are security implications with the current setup, as outlined in the issue, as the example reverse proxy configs does not secure Syncthing properly without modifications.

**Comment by calmh** [comment]
The docs specify a syncthing listening on the (default) localhost port 8384. This is not accessible from the outside except via the reverse proxy, which you control.
